### ISSUE TITLE
React 18 (RC 0) StrictMode rendering breaks ability to find node by it's draggable id

### ISSUE THREAD

**Issue by Malien** [atlassian/react-beautiful-dnd#2350]
### Expected behavior
React Beautiful DnD should work as advertised on the latest React version with StrictMode enabled (which is a common default nowadays)

### Actual behavior
Library fails to find draggable nodes with messaged logged (in development mode):
`Unable to find draggable with id: card-2`

When using the, now deprecated, ReactDOM.render API (instead of new ReactDOM.createRoot), React is forced to work in React 17 compatible manner, which eliminates the problem.

As I've found out, removing `<React.StrictMode />` wrapper resolved the issue, which means [React Beautiful DnD is not compatible with Strict Effects](https://github.com/reactwg/react-18/discussions/19)

> With the release of React 18, StrictMode gets an additional behavior that we call “strict effects” mode. When strict effects are enabled, React intentionally double-invokes effects (mount -> unmount -> mount) for newly mounted components. Like other strict mode behaviors, React will only do this for development builds.

### Steps to reproduce
1. `yarn add react@rc react-dom@rc react-beautiful-dnd`, 
2. 
```js 
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
)
```
3. Follow the guide precisely
4. \*profit\*

### Suggested solution?
I am not familiar with innerworkings of the library, but I suspect the way registrations `registry.draggable` are handled makes StrictMode incompatible (I've only looked into it in debugger for like 10 seconds). I suspect other registrations also fail

### What version of `React` are you using?
18.0.0-rc0

### What version of `react-beautiful-dnd` are you running?
13.1.2

### What browser are you using?
Safari 15.2 beta, Chrome 96

### Demo
https://codesandbox.io/s/vertical-list-forked-hd8ve?file=/index.js

**Comment by faradaytrs** [comment]
Same here, any plans to resolve this?

**Comment by 100terres** [comment]
You can follow the progress of this issue here: https://github.com/react-forked/dnd/issues/317

**Comment by Shahzad6077** [comment]
any updates on this issue?

**Comment by mjoycemilburn** [comment]
Hit this issue today too. 'Fixed' by removing the <React.StrictMode> components round the rendering of<App>in my index.js.

**Comment by enisdenjo** [comment]
Beware guys that StrictMode also helps you with preparing your development builds for production. Make sure to check whether the issue is present in production too.

**Comment by Danzo7** [comment]
any fix yet?

**Comment by MohamedH1998** [comment]
Same issue here also

**Comment by ShekibAziz** [comment]
same problem

**Comment by peterb0yd** [comment]
yup, same thing

**Comment by bhagi108** [comment]
going through same issue.

**Comment by Meligy** [comment]
The workaround here works OK for me: https://github.com/atlassian/react-beautiful-dnd/issues/2399#issuecomment-1167427762

**Comment by Xhale1** [comment]
If anyone is looking for a less hacky solution, I have a fork of this library [@hello-pangea/dnd](https://github.com/hello-pangea/dnd) which natively supports react 18 and react strict mode.

Just replace `react-beautiful-dnd` with `@hello-pangea/dnd` and you should be good to go 👍

**Comment by Kahitar** [comment]
> If anyone is looking for a less hacky solution, I have a fork of this library [@hello-pangea/dnd](https://github.com/hello-pangea/dnd) which natively supports react 18 and react strict mode.
> 
> Just replace `react-beautiful-dnd` with `@hello-pangea/dnd` and you should be good to go 👍

Is this going to be merged into react-beautiful-dnd eventually?

**Comment by RGBWang** [comment]
same problem

**Comment by olumide12-cell** [comment]
@Xhale1 I seems not to get it, replacing it means installing @hello-pangea/dnd](https://github.com/hello-pangea/dnd). won't it affect the functions?

**Comment by 100terres** [comment]
@Kahitar 

> Is this going to be merged into react-beautiful-dnd eventually?

I do not think it will. Atlassian do not plan to put more effort on this library. At least in the foreseeable future.

**Comment by 100terres** [comment]
@olumide12-cell 

> @Xhale1 I seems not to get it, replacing it means installing @hello-pangea/dnd](https://github.com/hello-pangea/dnd). won't it affect the functions?

I'm not certain to understand the question, but yes you should try to install `@hello-pangea/dnd` instead of `react-beautiful-dnd` and replace all the imports in your code.

`@hello-pangea/dnd` is a fork of `react-beautiful-dnd` and it supports react 18 with strict mode enabled.

**Comment by linxiaowu66** [comment]
focus on this issue

**Comment by arfanliaqat** [comment]
Ran into same problem

**Comment by WGTW** [comment]
It's been nearly a year, please fix this.

**Comment by shunk76** [comment]
This article is helpful to me.
https://medium.com/@wbern/getting-react-18s-strict-mode-to-work-with-react-beautiful-dnd-47bc909348e4

**Comment by Grawl** [comment]
I can confirm that solution from @shunk76 comment is working

**Comment by ThaJay** [comment]
I can confirm that the solution https://github.com/atlassian/react-beautiful-dnd/issues/2350#issuecomment-1276934082 is working.

**Comment by immortalt** [comment]
> This article is helpful to me. https://medium.com/@wbern/getting-react-18s-strict-mode-to-work-with-react-beautiful-dnd-47bc909348e4

Yes, this works.

**Comment by navinrangar** [comment]
Using ID that is the same for draggableId prop instead of index for key prop solved my problem.

**Comment by minhtrung0110** [comment]
After some searching, I came across [a very helpful snippet](https://github.com/atlassian/react-beautiful-dnd/issues/2399#issuecomment-1167427762) by [GiovanniACamacho](https://github.com/GiovanniACamacho), which was converted to TypeScript by [Meligy](https://github.com/Meligy) in that same issue, which resolved my problems and made me able to drag and drop again, with Strict Mode enabled.

`import { useEffect, useState } from "react";
import { Droppable, DroppableProps } from "react-beautiful-dnd";

export const StrictModeDroppable = ({ children, ...props }: DroppableProps) => {
  const [enabled, setEnabled] = useState(false);

  useEffect(() => {
    const animation = requestAnimationFrame(() => setEnabled(true));

    return () => {
      cancelAnimationFrame(animation);
      setEnabled(false);
    };
  }, []);

  if (!enabled) {
    return null;
  }

  return <Droppable {...props}>{children}</Droppable>;
};`

**Comment by checklist** [comment]
I switched to https://github.com/hello-pangea/dnd and all issues were solved.

**Comment by Chaobanh27** [comment]
> If anyone is looking for a less hacky solution, I have a fork of this library [@hello-pangea/dnd](https://github.com/hello-pangea/dnd) which natively supports react 18 and react strict mode.
> 
> Just replace `react-beautiful-dnd` with `@hello-pangea/dnd` and you should be good to go 👍

Tks man you helped me

**Comment by leikoilja** [comment]
Confirmed that it works on NextJS 13. [Here](https://github.com/atlassian/react-beautiful-dnd/issues/2399#issuecomment-1175638194) is a typescript version for the `Droppable` components compatible with strict mode

**Comment by SumanKisku** [comment]
> https://github.com/hello-pangea/dnd is a fork that solves this stickmode problem
> 
> The funniest thing is that the lib was working correctly and suddenly it stopped working, I'm using React 18.2.0, Next 14.0.2

Is it still causing problem?

**Comment by CarloshDevBR** [comment]
I didn't have any more problems

**Comment by dimple06varshney** [comment]
I also faced the same issue, as I tried to find out the difference between the code I wrote and the example code from the documentation, I found <React.StrictMode> is only when I commented out, that it works.

**Comment by evgar** [comment]
Run into same problem

**Comment by mskKote** [comment]
Run into same problem with Next.js

Set StrictMode to false and it's working + made rnd work only on client (works only after useEffect)

**Comment by nipunadodan** [comment]
> If anyone is looking for a less hacky solution, I have a fork of this library [@hello-pangea/dnd](https://github.com/hello-pangea/dnd) which natively supports react 18 and react strict mode.
> 
> Just replace `react-beautiful-dnd` with `@hello-pangea/dnd` and you should be good to go 👍

I'm using this library called  [@hello-pangea/dnd](https://github.com/hello-pangea/dnd) yet I still get this error. Any solutions? 

I'm using this on a Next.js 14 setup. And the issue occurs only on random occasions like it is the last element to move. However, the error reproduction is not consistent either.

**Comment by deepak-terse** [comment]
> If anyone is looking for a less hacky solution, I have a fork of this library [@hello-pangea/dnd](https://github.com/hello-pangea/dnd) which natively supports react 18 and react strict mode.
> 
> Just replace `react-beautiful-dnd` with `@hello-pangea/dnd` and you should be good to go 👍

I'm using this library in React and I'm still getting this error. For me the issue occurs on every server restart but it works correctly after a hot reload
